tensorflow/core/common_runtime/gpu/gpu_device.cc

```diff
@@ -254,6 +254,13 @@ class BaseGPUDevice::StreamGroupFactory {
       VLOG(2) << "Created nccl_stream[" << stream_group_within_gpu
               << "] = " << group->nccl;
 
+      // ROCm streams are lightweight and will not necessarily trigger device
+      // queue init until they are first used. For optimal performance,
+      // compute and nccl streams must be immediate siblings.
+      // Force underlying resource creation now.
+      group->compute->ThenWaitFor(group->nccl);
+      group->nccl->ThenWaitFor(group->compute);
```

Comment:
@jeffdaily the PR itself looks good to me. but should the comments be tuned? If I understand correctly both HIP streams and CUDA streams are lightweight so changes proposed here might also work for CUDA?